controlplane/kubeadm/internal/control_plane.go

```diff
@@ -253,6 +253,11 @@ func (c *ControlPlane) HasUnhealthyMachine() bool {
 	return len(c.UnhealthyMachines()) > 0
 }
 
+// HasHealthyMachineStillProvisioning returns true if any healthy machine in the control plane is still in the process of being provisioned.
+func (c *ControlPlane) HasHealthyMachineStillProvisioning() bool {
+	return len(c.Machines.Filter(collections.Not(collections.HasUnhealthyCondition), collections.And(collections.Not(collections.HasNode())))) > 0
```

Comment:
Just to confirm, this is checking if any of the machines available don't have a node associated AND are not marked as unhealthy (yet)?

Reply:
Yes
The idea is that we want to wait for all the machines to be provisioned before remediation, but we should preserve the capability to remediate a machine that fails to provision

Reply:
Improved the filter and added unit tests